### PR TITLE
Report semgrep cli failures to backend and receive fail open/close

### LINKED ISSUE #174: semgrep-action running semgrep with exit code >= 2 should be sent to some backend api
Users should be alerted of these unexpected failures in a noisy way

### PR DESCRIPTION
**Pull request by brendongo**

### CODE REVIEW

File: src/semgrep_agent/semgrep_app.py [hunk 1]
```diff
@@ -124,6 +125,31 @@ def download_rules(self) -> Tuple[Path, int]:
         rules_path.write_text(rules)
         return rules_path, len(parsed["rules"])
 
+    def report_failure(self, error: SemgrepError) -> int:
+        """
+        Send semgrep cli non-zero exit code information to server
```
Comment (chmccreery) [inline src/semgrep_agent/semgrep_app.py:130]: 🙏 thank you for the nice docstrings as usual. I gotta get back into this habit myself.

**Review by chmccreery — APPROVED** [review]
lgtm

File: src/semgrep_agent/main.py [hunk 2]
```diff
@@ -150,14 +152,30 @@ def main(
...
+        )
+    except SemgrepError as error:
+        print_sh_error_info(error.stdout, error.stderr, error.command, error.exit_code)
+
+        # If logged in handle exception
+        if sapp.is_configured:
+            exit_code = sapp.report_failure(error)
+            if exit_code == 0:
+                click.echo(
+                    f"Was going to exit with code {error.exit_code} but semgrep-app configuration is set to fail open so exiting with code 0.",
```
Comment (nbrahms) [inline src/semgrep_agent/main.py:172]:
```suggestion
                    f"Semgrep returned an error (code {error.exit_code}). However, this project's policy on {sapp.url} is configured to pass the build on Semgrep errors. This action will exit with return code 0.",
```

**Review by nbrahms — APPROVED** [review]

File: src/semgrep_agent/main.py (hunk 2 above)
Comment (brendongo) [reply src/semgrep_agent/main.py:172]: I'll create a follow up PR with this change

**Pull request merged**